components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -145,8 +147,10 @@ open class GleanInternalAPI internal constructor () {
         @Suppress("EXPERIMENTAL_API_USAGE")
         Dispatchers.API.flushQueuedInitialTasks()
 
-        // At this point, all metrics and events can be recorded.
-        ProcessLifecycleOwner.get().lifecycle.addObserver(gleanLifecycleObserver)
+        MainScope().launch {
```

Comment:
By looking at the docs and the [rationale for this](https://github.com/Kotlin/kotlinx.coroutines/issues/829), looks like we should declare `private val mainScope = MainScope()` in the `GleanInternalAPI` class and use `mainScope.launch` here. This follows the same pattern we've used for `CoroutineScope` in the dispatcher.

I don't know if that's *strictly* required, but I wouldn't want this to have surprising behaviours due to its short lived existence.

Reply:
Can you please explain your concerns about the `MainScope().launch` lifetime.  I read the issue you linked as the "rationale" and it is just in support of creating the MainScope() factory method, I don't see anything talking about lifetime issues.  

I see this same pattern being used in other places in Android Studio.